src/scripts/main.js

```diff
@@ -1,3 +1,14 @@
 'use strict';
 
-// write your code here
+const table = document.querySelector('table');
+
+const rows = table.rows;
+
+for (let i = 0; i < rows.length; i++) {
+  const row = rows[i];
+  const secondCell = row.cells[1];
+  const newCell = secondCell.cloneNode(true);
+  const lastCellIndex = row.cells.length - 1;
+
+  row.insertBefore(newCell, row.cells[lastCellIndex]);
```

Comment:
The logic for inserting the cloned cell is incorrect. The `insertBefore` method is used to insert the new cell before the last cell, but the index `lastCellIndex` refers to the last cell itself. This means the new cell will be inserted before the last cell, not at the end of the row. If the intention is to append the new cell at the end, consider using `appendChild` instead.